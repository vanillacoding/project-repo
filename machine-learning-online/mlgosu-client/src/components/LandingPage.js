import React from 'react';
import { Header, Message, Grid, Button, Input } from 'semantic-ui-react';
import GoogleLogin from 'react-google-login';

const LandingPage = (props) => {

  const responseGoogle = (res) => {
    props.setUserInfo(
      {
        googleId: res.profileObj.googleId,
        name: res.profileObj.name,
        email: res.profileObj.email
      }
    );
  };

  const handleChange = (e) => {
    props.setDriveUrl(e.target.value);
  };

  const handleClick = (e) => {
    e.preventDefault();
    props.buttonTrigger(true);
  };

  return (

    <Grid container style={{ padding: '3em 5em' }}>
      <Grid.Row>
        <Grid.Column textAlign='right' >
          { !props.userInfo ?
          <GoogleLogin
            render={ renderProps =>(
                <button
                  onClick={ renderProps.onClick }
                  disabled={ renderProps.disabled }>
                  Google
                </button>
              )}
            clientId = { process.env.REACT_APP_GOOGLE_ID }
            buttonText="Google"
            onSuccess= { responseGoogle }
            cookiePolicy={'single_host_origin'}
          />: <span>Welcome! { props.userInfo.name }</span> }
        </Grid.Column>
      </Grid.Row>

      <Grid.Row>
        <Grid.Column>
          <Header as='h3' dividing>
            Machine Learning Online
          </Header>
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column>
          <p>
            Machine Learning Online is a website that performs linear regression using Google's TensorFlow.js. Training performance depends on the GPU of your computer. In my case, it takes about 10 seconds to process approximately 20,000 sample data. I use Apple MacBook Pro 13-Inch (2017, Touch Bar). Therefore, if you handling very large amounts of data you should be very careful.
          </p>
        </Grid.Column>
      </Grid.Row>

      <Grid.Row>
        <Grid.Column onChange= { handleChange } >
          <Input placeholder='url' />
          { props.driveUrl.length < 15 ?
            <Button id='disabledBtn' disabled>
              Submit
            </Button>:
            <Button id='abledBtn' as='a' tabIndex='0' onClick={ handleClick }>
              Submit
            </Button> }
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column>
          <Message>
            <Header as='h4'>NOTICE</Header>
              <ol>
                <li>prepare csv dataset (local)</li>
                <li>upload to google drive (your account)</li>
                <li>right click > sharing link copy</li>
                <li>refer to the example below to modify the url</li>
              </ol>
              <ul>
                <li>open? => uc?export</li>
                <li>origin:
                    https://drive.google.com/open?=download&id=1j78fFH-IZUXnEmW0dNgOd9M2w64oU1zj
                </li>
                <li>fixed url:
                    https://drive.google.com/uc?export=download&id=1j78fFH-IZUXnEmW0dNgOd9M2w64oU1zj
                </li>
              </ul>
          </Message>
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column>
          <Header as='h4' dividing>
            SAMPLE DATASET
          </Header>
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column>
          <p>
            BTC, ETH, XLM Historical daily price data 17.03 ~ 20.05
            <li>https://drive.google.com/uc?export=download&id=1dypyzne0Brd5YfqfmUYwqa99QHkfziPs</li>
          </p>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  )
}

export default LandingPage;
