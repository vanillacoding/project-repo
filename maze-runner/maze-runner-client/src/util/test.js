import { NODE_STATUS } from '../constant';
import { createNodes } from './maze';

export function getNodeColIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[1];
}

export function getNodeRowIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[0];
}

export function getPrevNodeId(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length < 2) {
    throw new Error('Invalid node id');
  }

  const row = getNodeRowIndex(nodeId);
  const col = getNodeColIndex(nodeId);

  return `${Number(row)}-${Number(col) - 1}`;
}

export function createById(startNodeId, endNodeId) {
  const exampleBlocks = [
    [0, 0, 0, 1, 0, 0, 1, 0, 0, 1],
    [0, 1, 0, 1, 0, 1, 1, 0, 1, 1],
    [0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1],
    [0, 0, 0, 1, 0, 1, 1, 1, 0, 0],
    [0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 1, 1, 1, 1, 1, 0],
    [1, 1, 1, 0, 1, 0, 0, 0, 0, 0],
    [1, 0, 0, 0, 1, 0, 1, 1, 0, 1],
    [1, 0, 1, 0, 0, 0, 0, 1, 0, 0],
  ];

  const startNodeIndex = {
    col: getNodeColIndex(startNodeId),
    row: getNodeRowIndex(startNodeId),
  };
  const endNodeIndex = {
    col: getNodeColIndex(endNodeId),
    row: getNodeRowIndex(endNodeId),
  };

  exampleBlocks[startNodeIndex.row][startNodeIndex.col] = 3;
  exampleBlocks[endNodeIndex.row][endNodeIndex.col] = 4;

  const [widthCount, heightCount] = [
    exampleBlocks[0].length,
    exampleBlocks.length,
  ];

  const nodes = createNodes(widthCount, heightCount);
  const { byId } = nodes;

  exampleBlocks.forEach((row, rowIndex) =>
    row.forEach((number, colIndex) => {
      const nodeId = `${rowIndex}-${colIndex}`;

      switch (number) {
        case 0: {
          byId[nodeId].status = NODE_STATUS.UNVISITED;
          return;
        }
        case 1: {
          byId[nodeId].status = NODE_STATUS.WALL;
          return;
        }
        case 2: {
          byId[nodeId].status = NODE_STATUS.WEIGHTED;
          byId[nodeId].weight = 10;
          return;
        }
        case 3: {
          byId[nodeId].status = NODE_STATUS.START;
          return;
        }
        case 4: {
          byId[nodeId].status = NODE_STATUS.END;
          return;
        }
        case 5: {
          byId[nodeId].status = NODE_STATUS.MIDDLE;
          return;
        }
        default: {
          return;
        }
      }
    }),
  );

  return byId;
}

export default {
  getNodeColIndex,
  getNodeRowIndex,
  getPrevNodeId,
  createById,
};
