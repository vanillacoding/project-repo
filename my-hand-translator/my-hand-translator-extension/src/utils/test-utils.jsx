import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import userReducer from "../features/user/userSlice";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { user: userReducer }, preloadedState }),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

const user = {
  projectId: "translate-324702",
  translations: [
    {
      nanoId: "qWYBIfLPMsjP1LHNRqlFW",
      origin: "react",
      translated: "리액트",
      url: "vanilla",
      glossary: {
        react: "리액트",
      },
      createdAt: "2017-01-01",
    },
  ],
  glossary: {
    react: "리액트",
  },
  glossaryId: "6141bd1c2a11e5b51320bac1",
  isServerOn: false,
  name: "aidencoders",
  email: "aidencoders@gmail.com",
};

export { render, user };
