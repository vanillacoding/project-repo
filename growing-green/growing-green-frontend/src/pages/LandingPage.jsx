import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getAuth, signInWithPopup, GoogleAuthProvider } from 'firebase/auth';
import styled from 'styled-components';
import { FcGoogle } from 'react-icons/fc';
import { loginSuccess, logoutSuccess } from '../redux/modules/user';
import { getAllPlantsByUserId, updateAllPlant } from '../redux/modules/plants';
import { calculatePlantInfo } from '../utils/calcuatePlantInfo';

import AnimationText from '../components/AnimationText';
import TextButton from '../components/TextButton';
import PlantShelf from '../components/PlantShelf';
import ErrorBox from '../components/ErrorBox';

export default function LandingPage() {
  const history = useHistory();
  const dispatch = useDispatch();
  const { isLogin, error } = useSelector((state) => state.user);
  const { allPlants } = useSelector((state) => state.plants);
  const { weather } = useSelector((state) => state.environments);

  useEffect(() => {
    const user = localStorage.getItem('user');

    if (user) {
      dispatch(getAllPlantsByUserId());
    }
  }, [isLogin, dispatch]);

  useEffect(() => {
    const updatedAllPlants = [];
    const today = new Date();

    if (Object.keys(allPlants).length) {
      for (const id in allPlants) {
        const updatedPlant = calculatePlantInfo(allPlants[id], weather, today);

        updatedAllPlants.push(updatedPlant);
      }
    }

    dispatch(updateAllPlant(updatedAllPlants));
  }, [allPlants, dispatch]);

  if (error) {
    return <ErrorBox message={error} />;
  }

  async function loginWithGoogle() {
    try {
      const auth = getAuth();
      const {
        user: { displayName, email, photoURL },
      } = await signInWithPopup(auth, new GoogleAuthProvider());

      dispatch(
        loginSuccess({
          email,
          name: displayName,
          photoURL,
        }),
      );
    } catch {}
  }

  function onStartButtonClick() {
    const allPlantIds = Object.keys(allPlants);

    if (allPlantIds.length > 0) {
      return history.push(`/plants/${allPlantIds[0]}`);
    }

    history.push('/create');
  }

  function logout() {
    return dispatch(logoutSuccess());
  }

  function renderLoginButton() {
    return (
      <TextButton
        onClick={loginWithGoogle}
        variant="outline"
        size="large"
        color="white"
        label="L O G I N"
        icon={FcGoogle}
      />
    );
  }

  function renderStartButton() {
    return (
      <>
        <TextButton
          variant="outline"
          color="green"
          size="large"
          label="S T A R T"
          onClick={onStartButtonClick}
        />
        <LogoutText onClick={logout}>logout</LogoutText>
      </>
    );
  }

  return (
    <Container>
      <PlantShelf />
      <AnimationText>Touch the plant!</AnimationText>
      <ButtonWrapper>
        {isLogin ? renderStartButton() : renderLoginButton()}
      </ButtonWrapper>
    </Container>
  );
}

const Container = styled.div`
  position: relative;
  z-index: 1;
  border-radius: inherit;
  padding: 2rem;
  max-width: 100%;
  text-align: center;
`;

const ButtonWrapper = styled.div`
  display: inline-block;
`;

const LogoutText = styled.h3`
  font-size: 1.2em;
  font-weight: 300;
  margin: 1rem;

  &:hover {
    color: red;
    cursor: pointer;
  }
`;
