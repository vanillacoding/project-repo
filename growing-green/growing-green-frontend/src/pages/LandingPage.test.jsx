import React from 'react';
import { render } from '../test/test-utils';
import LandingPage from './LandingPage';

describe('<Landing>', () => {
  it('matches snapshot', () => {
    const { container } = render(<LandingPage />);
    expect(container).toMatchSnapshot();
  });

  it('should show login text on button', () => {
    const { getByRole } = render(<LandingPage />);
    expect(getByRole('button').textContent).toBe('L O G I N');
  });
});
