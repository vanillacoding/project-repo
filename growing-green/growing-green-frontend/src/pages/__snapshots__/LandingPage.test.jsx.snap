// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing> matches snapshot 1`] = `
<div>
  <div
    class="sc-kDTinF jWwnnS"
  >
    <div
      class="sc-dkPtRN hvIZeU"
    >
      <img
        alt="landscape image"
        class="sc-hKwDye kRHFfw"
        src="plant_shelf.png"
      />
      <img
        alt="logo text: Growing Green"
        class="sc-eCImPb hpeDWD"
        src="logo_text.png"
      />
      <img
        alt="moving plant image"
        class="sc-jRQBWg hfYcWn"
        src="moving_plant1.png"
      />
      <img
        alt="moving plant image"
        class="sc-gKclnd etgHhU"
        src="moving_plant2.png"
      />
      <img
        alt="moving plant image"
        class="sc-iCfMLu bxtLNE"
        src="moving_plant3.png"
      />
      <img
        alt="moving plant image"
        class="sc-furwcr ibbrex"
        src="moving_plant4.png"
      />
      <img
        alt="moving plant image"
        class="sc-pVTFL coAymM"
        src="moving_plant5.png"
      />
    </div>
    <h3
      class="sc-bdvvtL gIMoMb"
    >
      <span>
        T
      </span>
      <span>
        o
      </span>
      <span>
        u
      </span>
      <span>
        c
      </span>
      <span>
        h
      </span>
      <span>
         
      </span>
      <span>
        t
      </span>
      <span>
        h
      </span>
      <span>
        e
      </span>
      <span>
         
      </span>
      <span>
        p
      </span>
      <span>
        l
      </span>
      <span>
        a
      </span>
      <span>
        n
      </span>
      <span>
        t
      </span>
      <span>
        !
      </span>
    </h3>
    <div
      class="sc-iqseJM duXjpg"
    >
      <button
        class="sc-gsDKAQ bEMEza"
        color="white"
        label="L O G I N"
        type="button"
      >
        <svg
          enable-background="new 0 0 48 48"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 48 48"
          width="24"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#FFC107"
          />
          <path
            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
            fill="#FF3D00"
          />
          <path
            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
            fill="#4CAF50"
          />
          <path
            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
            fill="#1976D2"
          />
        </svg>
        L O G I N
      </button>
    </div>
  </div>
</div>
`;
